src/api/getAllData.tsx

```diff
@@ -0,0 +1,272 @@
+      let topicsModerateNum: number = 0;
+      let topicsLowNum: number = 0;
+
+      let topicsCritical: number[] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+      let topicsHigh: number[] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+      let topicsModerate: number[] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+      let topicsLow: number[] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+
+      for(let vulns of newVulnsFormatted){
+        let createdDate = new Date(vulns.createdAt)
+        let createdIndex = createdDate.getMonth()
+        let dismissedIndex = vulns.dismissedAt != null ? new Date(vulns.dismissedAt).getMonth() : -1
+
+        let today = new Date()
+        const vulnJSON = JSON.parse(JSON.stringify(vulns))
+        const vulnName = vulnJSON.packageName
+        if(vulns.dismissedAt){
+          offenses.add(vulnName)
+        }
+        else {
+          if(offenses.has(vulnName)) {
+            teamData.offenses += 1
+          }
+        }
+        if(today.getFullYear() - createdDate.getFullYear() == 0 || 
+          (today.getFullYear() - createdDate.getFullYear() == 1 && createdDate.getMonth() - today.getMonth() > 0)) {
+          if(vulns.severity == "CRITICAL"){
+
+            if (vulns.state == "OPEN"){
+              newRepo.critical += 1
+              teamData.vulnData.criticalNum += 1
+            }
+            
+            let i = createdIndex
+            do {
+              i %= 12
+              teamData.vulnData.critical[i] += 1
+              i += 1
+            } while(i != dismissedIndex && i != today.getMonth() + 1)
+
+            if(!seen.has(id)) {
+              if (vulns.state == "OPEN"){
+                orgData.vulnData.criticalNum += 1
+                topicsCriticalNum += 1
+              }
+
+              let i = createdIndex
+              do {
+                i %= 12
+                orgData.vulnData.critical[i] += 1
+                topicsCritical[i] += 1
+                i += 1
+              } while(i != dismissedIndex && i != today.getMonth() + 1) 
+            }
+          }
+          else if(vulns.severity == "HIGH"){
+
+            if (vulns.state == "OPEN"){
+              newRepo.high += 1
+              teamData.vulnData.highNum += 1
+            }
+
+            let i = createdIndex
+            do{
+              i %= 12
+              teamData.vulnData.high[i] += 1
+              i += 1
+            } while(i != dismissedIndex && i != today.getMonth() + 1)
+
+            if(!seen.has(id)) {
+              if (vulns.state == "OPEN"){
+                orgData.vulnData.highNum += 1
+                topicsHighNum += 1
+              }
+              let i = createdIndex
+              do {
+                i %= 12
+                orgData.vulnData.high[i] += 1
+                topicsHigh[i] += 1
+                i += 1
+              } while(i != dismissedIndex && i != today.getMonth() + 1) 
+            }
+          }
+          else if(vulns.severity == "MODERATE"){
```

Comment:
There is some code duplication here. It looks like these three `if` and `else if` blocks do the same thing. They just increase a different attribute depending on the severity. I'd suggest making a helper function to reduce code duplication.